### ISSUE TITLE
DEFAULT_METHOD_WHITELIST does not exist anymore

### ISSUE THREAD

**Issue by kparaju** [databricks/databricks-cli#636]
DEFAULT_METHOD_WHITELIST does not exist anymore and it causes the following issue with `dbx deploy`
```
│ /home/circleci/.pyenv/versions/3.9.16/lib/python3.9/site-packages/databricks │
│ _cli/sdk/api_client.py:106 in __init__                                       │
│                                                                              │
│   103 │   │   │   total=6,                                                   │
│   104 │   │   │   backoff_factor=1,                                          │
│   105 │   │   │   status_forcelist=[429],                                    │
│ ❱ 106 │   │   │   method_whitelist=set({'POST'}) | set(Retry.DEFAULT_METHOD_ │
│   107 │   │   │   respect_retry_after_header=True,                           │
│   108 │   │   │   raise_on_status=False # return original response when retr │
│   109 │   │   )                                                              │
╰──────────────────────────────────────────────────────────────────────────────╯
AttributeError: type object 'Retry' has no attribute 'DEFAULT_METHOD_WHITELIST'
```

This codebase [references](https://github.com/databricks/databricks-cli/blob/5c3879188322c0dfe94701db1ec8299bad7ccd5f/databricks_cli/sdk/api_client.py#LL106C56-L106C80
) `Retry.DEFAULT_METHOD_WHITELIST` but it doesn't exist anymore. It was [removed](https://github.com/urllib3/urllib3/commit/382ab32f23795c44faae83b4e8b18a16fb605a0a) from urllib3.

> Using 'Retry.DEFAULT_METHOD_WHITELIST' is deprecated and 
> will be removed in v2.0. Use 'Retry.DEFAULT_METHODS_ALLOWED' instead

Locking the version for urllib3 fixes the issue:
```
pip install dbx urllib3==1.26.15
```

**Comment by bigdatamoore** [comment]
I am getting the same error.

**Comment by jonsnowseven** [comment]
Me too. I solved it using `pip install 'urllib3<2'`.

Requests 2.30.0 release specifies possible breaking changes: https://requests.readthedocs.io/en/latest/community/updates/#id1

**Comment by chrimaho** [comment]
Same error for me too.

The error is occurring in the line 106 of the [databricks_cli/sdk/api_client.py](https://github.com/databricks/databricks-cli/blob/5c3879188322c0dfe94701db1ec8299bad7ccd5f/databricks_cli/sdk/api_client.py) file.

See:

https://github.com/databricks/databricks-cli/blob/5c3879188322c0dfe94701db1ec8299bad7ccd5f/databricks_cli/sdk/api_client.py#LL106C56-L106C80

More specifically, as noted in the CHANGELOG for [`urllip3`](https://github.com/urllib3/urllib3/tree/main) library, for version [`1.26.0`](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst#1260-2020-11-10):

> Deprecated `Retry` options `Retry.DEFAULT_METHOD_WHITELIST`, `Retry.DEFAULT_REDIRECT_HEADERS_BLACKLIST` and `Retry(method_whitelist=...)` in favor of `Retry.DEFAULT_ALLOWED_METHODS`, `Retry.DEFAULT_REMOVE_HEADERS_ON_REDIRECT`, and `Retry(allowed_methods=...)` (Pull #2000) **Starting in urllib3 v2.0: Deprecated options will be removed**

As the urllib3 package [`v2.0.0`](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst#200-2023-04-26) was released on 25/Apr/2023, these deprecated options have now been removed; as such, this error is now being thrown.

As @jonsnowseven mentiones, this could be resolved by forcing the install of `urllib3<2`. But that should only be a temporary workaround. Long term, we need to change the [`ApiClient`](https://github.com/databricks/databricks-cli/blob/5c3879188322c0dfe94701db1ec8299bad7ccd5f/databricks_cli/sdk/api_client.py#LL92C5-L92C5) class to no longer reference the deprecated attributes from [`urllip3`](https://github.com/urllib3/urllib3/tree/main).

**Comment by shreyas-goenka** [comment]
Cutting a new path release of the cli and dbx before closing the issue

**Comment by rootd00d** [comment]
This issue seems to have broken the "DevOps for Azure Databricks" ADO extension.  Anyway we can accelerate a fix there as well?

**Comment by shreyas-goenka** [comment]
Hi @rootd00d , The ADO extension is owned by Microsoft and seems to not have any active support. Anyways, once the databricks cli fix is out, it should take care of the extension as well once you upgrade databricks-cli dependency to the latest patch version

**Comment by pietern** [comment]
We just released version 0.17.7 to PyPI which includes a fix.
